File: php/features.php
<?php
declare(strict_types=1);

// CompanySearch SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CompanySearchFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CompanySearchBaseFeature();
            case "test":
                return new CompanySearchTestFeature();
            default:
                return new CompanySearchBaseFeature();
        }
    }
}

File: py/features.py
# CompanySearch SDK feature factory

from feature.base_feature import CompanySearchBaseFeature
from feature.test_feature import CompanySearchTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CompanySearchBaseFeature(),
        "test": lambda: CompanySearchTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
